metpy/calc/thermo.py

```diff
@@ -1608,7 +1612,12 @@ def _isen_iter(iter_log_p, isentlevs_nd, ka, a, b, pok):
 
     # Change when Python 2.7 no longer supported
     # Pull out keyword arguments
-    tmpk_out = kwargs.pop('tmpk_out', False)
+    temperature_out = kwargs.get('tmpk_out')
+    if temperature_out:
```

Comment:
Needs to be: `if temperature_out is not None` so that we warn when `tmpk_out=False` is passed.